pkg/controller/registrationservice/registrationservice_controller.go

```diff
@@ -141,7 +141,7 @@ func (r *ReconcileRegistrationService) Reconcile(request reconcile.Request) (rec
 	// create all objects that are within the template, and update only when the object has changed.
 	var updated []string
 	for _, toolchainObject := range toolchainObjects {
-		createdOrUpdated, err := cl.CreateOrUpdateObject(toolchainObject.GetRuntimeObject(), false, regService)
+		createdOrUpdated, err := cl.ApplyObject(toolchainObject.GetRuntimeObject())
```

Comment:
you didn't add the `regService` as the owner of the object

Reply:
ah right! Fixed in 6af0032 along with some refactoring on the ServiceAccount and ConfigMap assertions